### PR TITLE
Make account address create/update and default address set a live components

### PR DESCRIPTION
**Pull request by TheMilek**
| Q               | A
|-----------------|-----
| Branch?         | bootstrap-shop
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| License         | MIT

### CODE REVIEW

File: src/Sylius/Bundle/ShopBundle/Resources/config/services/twig/component.xml [hunk 1]
```diff
@@ -306,5 +306,33 @@
                 template="@SyliusShop/account/change_password/form.html.twig"
             />
         </service>
+
+        <service
+            id="sylius_shop.twig.component.set_default_address.form"
+            class="Sylius\Bundle\ShopBundle\Twig\Component\Account\SetDefaultAddress\FormComponent"
```
Comment (mpysiak) [inline src/Sylius/Bundle/ShopBundle/Resources/config/services/twig/component.xml]:
```suggestion
            class="Sylius\Bundle\ShopBundle\Twig\Component\Account\DefaultAddress\FormComponent"
```
🤔

**Review by mpysiak — APPROVED** [review]

File: features/shop/account/customer_account/address_book/editing_address.feature [hunk 2]
```diff
@@ -39,7 +39,7 @@ Feature: Making changes in existing addresses
         And it should contain "Liverpool"
         And it should contain "GBA-20B"
 
-    @todo @ui @javascript @api
+    @ui @javascript @api
```
Comment (Wojdylak) [inline features/shop/account/customer_account/address_book/editing_address.feature]:
```suggestion
    @api @ui @javascript
```

File: src/Sylius/Bundle/ShopBundle/Twig/Component/Account/Address/FormComponent.php [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    /** @param class-string $formClass */
+    public function __construct(
+        private readonly FormFactoryInterface $formFactory,
+        private readonly string $formClass,
+    ) {
+    }
+
+    protected function instantiateForm(): FormInterface
+    {
+        return $this->formFactory->create($this->formClass, $this->address);
```
Comment (Wojdylak) [inline src/Sylius/Bundle/ShopBundle/Twig/Component/Account/Address/FormComponent.php]:
```suggestion
        return $this->formFactory->create($this->formClass, $this->address, ['add_build_address_form_subscriber' => false]);
```
It should be handled the same way as the AdminBundle and also needs to be added to the route configuration

File: src/Sylius/Bundle/ShopBundle/Resources/config/services/twig/component.xml (hunk 1 above)
Comment (Wojdylak) [reply src/Sylius/Bundle/ShopBundle/Resources/config/services/twig/component.xml]: or
```suggestion
            class="Sylius\Bundle\ShopBundle\Twig\Component\Account\Address\DefaultFormComponent"
```

File: src/Sylius/Bundle/ShopBundle/Resources/config/services/twig/component/account.xml [hunk 4]
```diff
@@ -52,5 +52,33 @@
...
+        <service
+            id="sylius_shop.twig.component.set_default_address.form"
+            class="Sylius\Bundle\ShopBundle\Twig\Component\Account\Address\DefaultFormComponent"
+        >
+            <argument type="service" id="form.factory" />
+            <argument>Sylius\Bundle\ShopBundle\Form\Type\CustomerDefaultAddressType</argument>
+
+            <tag
+                name="sylius.live_component"
+                key="sylius_shop:account:set_default_address:form"
```
Comment (Wojdylak) [inline src/Sylius/Bundle/ShopBundle/Resources/config/services/twig/component/account.xml]:
```suggestion
                key="sylius_shop:account:address:address:default_form"
```

File: src/Sylius/Bundle/ShopBundle/Resources/config/services/twig/component/account.xml [hunk 5]
```diff
@@ -52,5 +52,33 @@
                 template="@SyliusShop/account/change_password/form.html.twig"
             />
         </service>
+
+        <service
+            id="sylius_shop.twig.component.set_default_address.form"
```
Comment (Wojdylak) [inline src/Sylius/Bundle/ShopBundle/Resources/config/services/twig/component/account.xml]:
```suggestion
            id="sylius_shop.twig.component.account.address.default_form"
```

File: src/Sylius/Bundle/ShopBundle/Resources/config/services/twig/component/account.xml [hunk 6]
```diff
@@ -52,5 +52,33 @@
...
+
+            <tag
+                name="sylius.live_component"
+                key="sylius_shop:account:set_default_address:form"
+                template="@SyliusShop/account/address_book/set_as_default.html.twig"
+            />
+        </service>
+
+        <service
+            id="sylius_shop.twig.component.address.form"
```
Comment (Wojdylak) [inline src/Sylius/Bundle/ShopBundle/Resources/config/services/twig/component/account.xml]:
```suggestion
            id="sylius_shop.twig.component.account.address.form"
```

File: src/Sylius/Bundle/ShopBundle/Resources/config/services/twig/component/account.xml [hunk 7]
```diff
@@ -52,5 +52,33 @@
                 template="@SyliusShop/account/change_password/form.html.twig"
             />
         </service>
+
+        <service
+            id="sylius_shop.twig.component.account.default_address.form"
```
Comment (GSadee) [inline src/Sylius/Bundle/ShopBundle/Resources/config/services/twig/component/account.xml]:
```suggestion
            id="sylius_shop.twig.component.account.address.default_form"
```

**Review by GSadee — APPROVED** [review]

**Review by Wojdylak — APPROVED** [review]

**Review by mpysiak — APPROVED** [review]

**Pull request merged**